### PR TITLE
Rotation keep history

### PR DESCRIPTION
**Pull request by FedericoCeratto**
Add basic testing, cleanups

### CODE REVIEW

File: af/analysis/tests/test_rotation.py [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+import datetime
+from unittest.mock import Mock, create_autospec, call
+
+import pytest  # debdeps: python3-pytest
+
+import rotation as ro
+
+
+@pytest.fixture(autouse=True, scope="session")
+def mock_everything():
```
Comment (bassosimone) [inline af/analysis/tests/test_rotation.py:14]: I love the name of this function! 🤣

**Review by bassosimone — APPROVED** [review]
🐳

File: af/analysis/rotation.py [hunk 2]
```diff
@@ -24,6 +30,7 @@
     `ipaddr` IPv4,
     `ipv6addr` IPv6,
     `draining_at` Nullable(DateTime('UTC')),
+    `destroyed_at` Nullable(DateTime('UTC')),
     `sign` Int8
 )
 ENGINE = CollapsingMergeTree(sign)
```
Comment (hellais) [inline af/analysis/rotation.py:36]: If we are effectively always using sign = 1, why don't we at that point just use a ReplacingMergeTree?

Comment (FedericoCeratto) [reply af/analysis/rotation.py:36]: In the past we wanted to control deletions using CollapsingMergeTree. In future versions we might switch database entirely as discussed.

**Pull request merged**